sos/report/plugins/ceph_rgw.py

```diff
@@ -0,0 +1,100 @@
+
+from sos.report.plugins import Plugin, RedHatPlugin, UbuntuPlugin
+from socket import gethostname
+import os
+
+
+class CephRGW(Plugin, RedHatPlugin, UbuntuPlugin):
+
+    short_desc = 'CEPH rgw'
+
+    plugin_name = 'ceph_rgw'
+    profiles = ('storage', 'virt', 'container')
+    containers = ('ceph-rgw.*',)
+    ceph_hostname = gethostname()
+
+    services = (
+        'ceph-radosgw@*',
+    )
+
+    def check_enabled(self):
+        # first check services running
+        if super(CephRGW, self).check_enabled():
+            return True
+
+        # also check non empty folders in /var/lib/ceph/radosgw
+        rgw_folder = '/var/lib/ceph/radosgw'
+        if os.path.exists(rgw_folder) and os.path.isdir(rgw_folder):
+            inner_files = os.listdir(rgw_folder)
+            for item in inner_files:
+                full_path = rgw_folder + "/" + item
+                if(os.path.isdir(full_path)):
+                    # verify the folder has files
+                    if not os.listdir(full_path):
+                        continue
+                    else:
+                        return True
+        return False
+
+    def setup(self):
+        # all_logs = self.get_option("all_logs")
+
+        self.add_file_tags({
+            '/var/log/ceph/ceph-client.rgw*.log': 'ceph_rgw_log',
+        })
+
+        self.add_copy_spec([
+                "/var/log/ceph/ceph-client.rgw*.log",
+        ])
+
+        # stub for now with just one
+        self.add_cmd_output([
+            "rados df",
+        ])
+
+        # may not be needed if no ceph rgw like commands.
+        # Are there any?
+        # ceph_cmds = [
+        # ]
+
+        # self.add_cmd_output([
+        #    "ceph %s --format json-pretty" % s for s in ceph_cmds
+        # ], subdir="json_output", tags="insights_ceph_health_detail")
+
+        self.add_forbidden_path([
+            "/etc/ceph/*keyring*",
+            "/var/lib/ceph/*keyring*",
+            "/var/lib/ceph/*/*keyring*",
+            "/var/lib/ceph/*/*/*keyring*",
+            "/var/lib/ceph/osd",
+            "/var/lib/ceph/mon",
+            # Excludes temporary ceph-osd mount location like
+            # /var/lib/ceph/tmp/mnt.XXXX from sos collection.
+            "/var/lib/ceph/tmp/*mnt*",
+            "/etc/ceph/*bindpass*"
+        ])
+
+        # If containerized, run commands in containers
+        # what happens if the container is named something else?
+        # containers_list = self.get_all_containers_by_regex("ceph-rgw*")
+        # if containers_list:
+        #    for container in containers_list:
+        #        self.add_cmd_output([
+        #            self.fmt_container_cmd(container[1], "ceph %s" % s)
+        #            for s in ceph_cmds
+        #        ])
+        #        break
+        # Not containerized
+        # else:
+        #   self.add_cmd_output([
+        #       "ceph %s" % s for s in ceph_cmds
+        #    ])
```

Comment:
If this isn't relevant to the plugin, let's remove this entirely rather than comment it out. Same for lines 62-69 above.

Reply:
I will keep this and in fact uncomment it. Instead of "ceph xxx" commands, I plan to collect "rados xxx" commands in the rgw plugi, like for eg "rados lspools", etc. If they might need the "fmt_container_cmd()" invokation instead of simply invoking through a "self.add_cmd_output()" call, this loop will be changed accordingly. I'm trying to understand why "some" ceph commands are added directly using "add_cmd_output() (without checking if its a container) while some need the fmt_container_cmd if its running in a container. Once I resolve that, I'll update this code accordingly.

Reply:
ack.